pennylane/hf/molecule.py

```diff
@@ -0,0 +1,153 @@
+    """
+
+    def __init__(
+        self,
+        symbols,
+        coordinates,
+        charge=0,
+        mult=1,
+        basis_name="sto-3g",
+        active_electrons=None,
+        active_orbitals=None,
+        l=None,
+        alpha=None,
+        coeff=None,
+        r=None,
+    ):
+
+        if basis_name not in ["sto-3g", "STO-3G"]:
+            raise ValueError("The only supported basis set is 'sto-3g'.")
+
+        if set(symbols) - set(atomic_numbers):
+            raise ValueError(f"Atoms in {set(symbols) - set(atomic_numbers)} are not supported.")
+
+        self.symbols = symbols
+        self.coordinates = coordinates
+        self.charge = charge
+        self.mult = mult
+        self.basis_name = basis_name
+
+        self.n_basis, self.basis_data = mol_basis_data(self.basis_name, self.symbols)
+
+        if l is None:
+            l = [i[0] for i in self.basis_data]
+
+        if alpha is None:
+            alpha = [pnp.array(i[1], requires_grad=False) for i in self.basis_data]
+
+        if coeff is None:
+            coeff = [pnp.array(i[2], requires_grad=False) for i in self.basis_data]
+
+        if r is None:
+            r = sum([[self.coordinates[i]] * self.n_basis[i] for i in range(len(self.n_basis))], [])
+
+        self.l = l
+        self.alpha = alpha
+        self.coeff = coeff
+        self.r = r
+
+        self.basis_set = generate_basis_set(self.l, self.alpha, self.coeff, self.r)
+        self.n_orbitals = len(self.l)
+
+        self.nuclear_charges = generate_nuclear_charges(self.symbols)
+
+        self.n_electrons = sum(np.array(self.nuclear_charges)) - self.charge
+        self.core, self.active = qml.qchem.active_space(
+            self.n_electrons,
+            self.n_orbitals,
+            active_electrons=active_electrons,
+            active_orbitals=active_orbitals,
+        )
+
+
+def generate_basis_set(l, alpha, coeff, r):
+    r"""Generate a set of basis function objects.
+
+    Args:
+        l list((tuple[int])): angular momentum numbers of the basis function.
+        alpha list((array(float))): exponents of the Gaussian functions forming basis functions
+        coeff list((array(float))): coefficients of the contracted Gaussian functions
+        r list((array(float))): positions of the Gaussian functions forming the basis functions
+
+    Returns:
+        list(BasisFunction): list containing a set of basis function objects.
+
+    **Example**
+
+    >>> l = [(0, 0, 0), (0, 0, 0)]
+    >>> exponents = [[3.42525091, 0.62391373, 0.1688554], [3.42525091, 0.62391373, 0.1688554]]
+    >>> coefficients = [[0.15432897, 0.53532814, 0.44463454], [0.15432897, 0.53532814, 0.44463454]]
+    >>> centers = [[0.0, 0.0, -0.694349], [0.0, 0.0, 0.694349]]
+    >>>  basis_set = generate_basis_set(l, exponents, coefficients, centers)
+    >>> print(basis_set)
+    [<molecule.BasisFunction object at 0x7f7566db2910>, <molecule.BasisFunction object at 0x7f7566db2a30>]
```

Comment:
```suggestion
    >>> basis_set = generate_basis_set(l, exponents, coefficients, centers)
    >>> print(basis_set)
    [<molecule.BasisFunction object at 0x7f7566db2910>, <molecule.BasisFunction object at 0x7f7566db2a30>]
```

This example is maybe not so intuitive, since exponents, centers, basis functions are never explained!

Also, could the `print()` be improved to show the details regarding the BasisFunction objects?